### PR TITLE
Allow minus sign at the start of a label.

### PR DESCRIPTION
**Pull request by bratseth**

### CODE REVIEW

**Comment by lesters** [comment]
@bratseth @havardpe Are there any implications for the backend here?

**Comment by havardpe** [comment]
yes, but I am working on supporting more relaxed labels in the back-end as well

File: vespajlib/src/main/java/com/yahoo/tensor/TensorAddress.java [hunk 1]
```diff
@@ -163,7 +163,7 @@ public String toString() {
     /** Supports building of a tensor address */
     public static class Builder {
 
-        private Pattern identifierPattern = Pattern.compile("[A-Za-z0-9_]+");
+        private Pattern identifierPattern = Pattern.compile("[-,A-Za-z0-9_@]([A-Z,a-z0-9_@$])*");
```
Comment (havardpe) [inline vespajlib/src/main/java/com/yahoo/tensor/TensorAddress.java]: I would suggest renaming this one now, sice we do not want it to be confused with identifiers

**Comment by havardpe** [comment]
this will allow labels to contain anything in the back-end (including '-'):
https://github.com/vespa-engine/vespa/pull/11524

**Review by lesters — APPROVED** [review]

File: vespajlib/src/main/java/com/yahoo/tensor/TensorAddress.java (hunk 1 above)
Comment (bratseth) [reply vespajlib/src/main/java/com/yahoo/tensor/TensorAddress.java]: Good point; done

**Review by baldersheim — APPROVED** [review]

**Pull request merged**